app/controllers/chat_channels_controller.rb

```diff
@@ -48,7 +48,7 @@ def create
     auto_join_users = ActiveRecord::Type::Boolean.new.deserialize(params[:auto_join_users]) || false
 
     chat_channel =
-      ChatChannel.create!(
+      ChatChannel.type_for(chatable).create!(
```

Comment:
What about doing this API:


```ruby
chatable.create_chat_channel!(
  name: params[:name],
  description: params[:description]
)
```

?

Reply:
I think this and the other suggestion would be more convenient too, even if we have to reopen the class via the plugin API to do it.